src/pykeen/sampling/negative_sampler.py

```diff
@@ -58,7 +58,7 @@ def __init__(
             filterer,
             pos_kwargs=filterer_kwargs,
             mapped_triples=triples_factory.mapped_triples,
-        ) if filtered else None
+        ) if filterer is not None or filtered else None
```

Comment:
@mberr this is an important bug fix! without it, the filtering actually was never tested